src/utilities/theme/utils.ts

```diff
@@ -1,115 +1,779 @@
+        hue,
+        saturation,
+        lightness: isLightTheme ? 98 : 2,
+      }),
+      warningText: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 30 : 70,
+      }),
+    };
+  }
+
+  function highlightColors(baseColor: string) {
+    const hslBaseColor = colorToHsla(baseColor) as HSLAColor;
+    const {hue, saturation} = hslBaseColor;
 
-function parseColors([baseName, colors]: [string, ColorsToParse]): string[][] {
-  const keys = Object.keys(colors);
-  const colorPairs = [];
-  for (let i = 0; i < keys.length; i++) {
-    colorPairs.push([constructColorName(baseName, keys[i]), colors[keys[i]]]);
+    return {
+      highlight: hslToString(hslBaseColor),
+      highlightDivider: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 58 : 42,
+      }),
+      highlightIcon: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 58 : 42,
+      }),
+      highlightSurface: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 88 : 12,
+      }),
+      highlightSurfaceMuted: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 98 : 2,
+      }),
+      highlightText: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 98 : 2,
+      }),
+    };
+  }
 
-    if (needsVariant(baseName)) {
-      const hslColor = colorToHsla(colors[keys[i]]);
+  function successColors(baseColor: string) {
+    const hslBaseColor = colorToHsla(baseColor) as HSLAColor;
+    const {hue, saturation} = hslBaseColor;
 
-      if (typeof hslColor === 'string') {
-        return colorPairs;
-      }
+    return {
+      success: hslToString(hslBaseColor),
 
-      const rgbColor = hslToRgb(hslColor);
+      // Pulled this 35 lightness value from a dark mode playground for an icon
+      successDivider: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 25 : 35,
+      }),
 
-      if (isLight(rgbColor)) {
-        colorPairs.push(...setTheme(hslColor, baseName, keys[i], 'light'));
-      } else {
-        colorPairs.push(...setTheme(hslColor, baseName, keys[i], 'dark'));
-      }
-    }
+      // Pulled this 35 lightness value from a dark mode playground for an icon
+      successIcon: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 25 : 35,
+      }),
+      successSurface: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 88 : 12,
+      }),
+      successSurfaceMuted: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 98 : 2,
+      }),
+      successText: hslToString({
+        hue,
+        saturation,
+        lightness: isLightTheme ? 15 : 85,
+      }),
+    };
   }
+}
+
+// TODO: Type name here is weird, we should either really type this so only camelCase key names are valid in the argument
+// or use a more generic name for this type
```

Comment:
👍

Reply:
I don't think you can make typescript say "a string that fits a particular format", the closest you can do is a set of literal types `type foo = 'a' | 'b' | 'c';` but I don't think you want to write out every possible value here.

You might also be interested in nominal typing: https://basarat.gitbooks.io/typescript/docs/tips/nominalTyping.html as a mechanism to say "this is just a string but we've already checked that it is this particular type of string